### PR TITLE
hide attention lstm fuse

### PR DESCRIPTION
**Pull request by Superjomn**
- hide the special case in attention lstm fuse pass.
- make the AnalysisConfig more general.

### CODE REVIEW

File: paddle/fluid/inference/api/paddle_inference_api.h [hunk 1]
```diff
@@ -204,10 +204,11 @@ struct AnalysisConfig : public NativeConfig {
     kExclude   // Specify the disabled passes in `ir_passes`.
   };
 
+  // Determine whether to perform graph optimization.
   bool enable_ir_optim = true;
+  // Manually determine the IR passes to run.
   IrPassMode ir_mode{IrPassMode::kExclude};
-  // attention lstm fuse works only on some specific models, disable as default.
-  std::vector<std::string> ir_passes{"attention_lstm_fuse_pass"};
+  std::vector<std::string> ir_passes;
```
Comment (panyx0718) [inline paddle/fluid/inference/api/paddle_inference_api.h:211]: my feeling is that you don't need the above 2 fields now.
I think setting these 2 fields have undefined behavior now.

Comment (luotao1) [inline paddle/fluid/inference/api/paddle_inference_api.h:211]: https://github.com/PaddlePaddle/Paddle/blob/1ab7b551625bf90d96840773c12fecdb53f878ce/paddle/fluid/inference/tests/api/analyzer_rnn1_tester.cc#L159
这句可以去掉了

**Review by panyx0718 — APPROVED** [review]

**Review by tensor-tang — COMMENTED** [review]
如果这里改掉之后，https://github.com/PaddlePaddle/Paddle/pull/13553 这个pr的怎么默认disable？

**Comment by Superjomn** [comment]
默认会根据模型自动开启或者disable attentionlstm fuse. @tensor-tang

**Comment by tensor-tang** [comment]
我是指embedding lstm那个怎么默认disable。原先的话只需要参考Attention lstm这个做法。

**Comment by Superjomn** [comment]
可以给 pass 加一个 bool 的 attr作为开关，默认不设置或者false，需要用的话再设置为 true。 @tensor-tang

**Pull request merged**